  public void testCheckThrowable_unsupportedThrowableConstructor_throwsIllegalArgumentException() {
    String message =
        String.format(
            "Thrown exception <%s> must have a no-argument constructor",
            StringException.class.getName());

    try {
      CheckedProviders.throwing(FooCheckedProvider.class, StringException.class);
      fail("Expected an exception to be thrown");
    } catch (IllegalArgumentException e) {
      assertThat(e).hasMessageThat().isEqualTo(message);
      assertWithMessage(String.format("exception <%s> with cause", e))
          .that(e.getCause())
          .isInstanceOf(NoSuchMethodException.class);
    }
  }

